daemon/api_connections_test.go

```diff
@@ -279,6 +286,67 @@ func (s *apiSuite) TestConnectionsBySnapName(c *check.C) {
 	})
 }
 
+func (s *apiSuite) TestConnectionsMissingPlugSlotFilteredOut(c *check.C) {
+	restore := builtin.MockInterface(&ifacetest.TestInterface{InterfaceName: "test"})
+	defer restore()
+
+	s.daemon(c)
+
+	s.mockSnap(c, consumerYaml)
+	s.mockSnap(c, producerYaml)
+
+	for _, missingPlugOrSlot := range []string{"consumer:plug2 producer:slot", "consumer:plug producer:slot2"} {
+		s.testConnectionsConnected(c, "/v2/connections?snap=producer", map[string]interface{}{
+			"consumer:plug producer:slot": map[string]interface{}{
+				"interface": "test",
+			},
+			missingPlugOrSlot: map[string]interface{}{
+				"interface": "test",
+			},
+		}, []string{"consumer:plug producer:slot"}, map[string]interface{}{
```

Comment:
Was a bit hard to spot the repo connections list.
```suggestion
		}, 
		[]string{"consumer:plug producer:slot"}, 
		map[string]interface{}{
```